### ISSUE TITLE
Add support for seamless reloads

### ISSUE THREAD

**Issue by kustodian** [iuscommunity-pkg/haproxy18u#5]
HAProxy 1.8 introduced support for seamless reloads which is currently not working when HAProxy is started with systemd (which is our case). Of course regular reloads work without any issues. When seamless reloads is fixed in HAProxy this package should support them out of the box.

To add support for seamless reloads the only thing that would need to be changed is one line in `haproxy.cfg`:

```diff
-    stats socket /var/lib/haproxy/stats
+    stats socket /var/lib/haproxy/stats expose-fd listeners
```
I'll update this issue once seamless reloads are fixed.

**Comment by carlwgeorge** [comment]
Is there an upstream bug for this?

**Comment by kustodian** [comment]
Not that I know. Tbh I'm not even sure where is the HAProxy bug tracker.

There is a forum discussion thread and a mailing list thread:
* https://discourse.haproxy.org/t/seamless-reloads-dont-work-with-systemd/1954/2
* https://www.mail-archive.com/haproxy@formilux.org/msg28775.html

**Comment by b-harper** [comment]
@kustodian,

I am just getting some time to read those mailing list and forum threads.  Either one of them directly mentions that your recommendation addresses this issue. I did find this [commit][0] that supports your recommendation.  It does appear we need to wait for the HAProxy devs to fix a bug.

Thanks for bringing to our attention.  We look forward to hear back when the bug has been fixed.


[0]: http://git.haproxy.org/?p=haproxy.git;a=commitdiff;h=190bbb8f53e210d1cc997bc70ac1fd131c6abbe5;hp=eb3e276d3911c4c95d1e2f270bab354440378701
